### ISSUE TITLE
console warning: cycle-plugin.js:49 Cyclic next detected

### ISSUE THREAD

**Issue by wardbell** [cartant/rxjs-spy#41]
Worrying console message
```
cycle-plugin.js:49 Cyclic next detected; type = subject; value = Scroll(anchor: 'null', position: 'null'); subscribed at
RouterScroller.push../node_modules/@angular/router/fesm5/router.js.RouterScroller.createScrollEvents()@http://localhost:5009/vendor.js:73846:35
RouterScroller.push../node_modules/@angular/router/fesm5/router.js.RouterScroller.init()@http://localhost:5009/vendor.js:73841:46
RouterInitializer.push../node_modules/@angular/router/fesm5/router.js.RouterInitializer.bootstrapListener()@http://localhost:5009/vendor.js:74223:24
{anonymous}()@http://localhost:5009/vendor.js:43425:56
```
Evidently something going on in Angular router. _Not asking you to fix it_. 

I am asking what the warning is telling me and what someone (the Angular team?) should do about it.

Trying this promising library for the first time with angular@^6.1.9 and rxjs@^6.3.3

**Comment by cartant** [comment]
The warning is shown when a call to `next` effects a cycle. E.g. if subject B is subscribed to subject B and subject A is subscribed to subject B, calling `next` on either subject will effect a cycle. If it's not a situation that the caller accounts for, it's a problem.

If you don't want the warning and are confident that any cyclical calls are dealt with, you can not use that particular plugin.

**Comment by cartant** [comment]
I suppose it would make sense to have a number-of-cycles threshold that higher than the current threshold of one.

**Comment by wardbell** [comment]
The _explanation_ is a boon. Adding to the docs would be great. 

I owe you a PR for that perhaps.

**Comment by cartant** [comment]
From memory, it was something that was added well after I'd written what documentation there is. I have vague memories of helping someone debug a problem and writing the plugin to try to make it a little easier and automated in future.

I have a few things to do with this project, but at the moment, things are beyond busy at the office. Hopefully, it will calm down soon and I can spend some more time on this project and its associated DevTools project.

**Comment by SerkanSipahi** [comment]
@cartant i get the same warning and im hoping it disappear with the cycleTreshold option which was added in https://github.com/cartant/rxjs-spy/commit/0e10d3fcf1e1c035f0f1dfab5cf09b072d9f9035. Could you may deploy a new version of rxjs-spy to that we can consume it with npm. Thank you.

FYI: You explained why this warning is displayed but i didnt understand it. Could you explain it easier?

**Comment by cartant** [comment]
> Could you may deploy a new version of rxjs-spy so that we can consume it with npm.

No. I'm not inclined to do that. Soon, I'm going to release `rxjs-devtools`. It will have some new features and will replace `rxjs-spy`. The changes you mention were reverted and the work I was doing was moved to a new repo - which will also include a Chrome DevTools extension.

> Could you explain it easier?

Not really. It means there is a feedback loop. That is, when a source observable emits, something happens in a listener that effects another synchronous emission from the source observable.

**Comment by rattkin** [comment]
I found this as well. Looking forward to chrome devtools extension!

**Comment by simonmulser** [comment]
currently (`7.5.1`) it is possible to deactive the plugin with `spy.unplug(spy.find(CyclePlugin));`

**Comment by msorens** [comment]
Anyone have ideas why @simonmulser 's suggestion does not work for me in 7.5.1?
```
> spy.unplug(spy.find(CyclePlugin))
Uncaught ReferenceError: CyclePlugin is not defined
    at <anonymous>:1:21
```

**Comment by bschantz** [comment]
@msorens The console environment won't have any knowledge of `CyclePlugin` -- it needs to be in code:
```ts
import { create } from 'rxjs-spy';
import { CyclePlugin } from 'rxjs-spy/plugin';
const spy = create();
spy.unplug(spy.find(CyclePlugin));
```
